redbot/cogs/audio/core/commands/controller.py

```diff
@@ -242,9 +248,12 @@ async def command_prev(self, ctx: commands.Context):
         """Skip to the start of the previously played track."""
         if not self._player_check(ctx):
             return await self.send_embed_msg(ctx, title=_("Nothing playing."))
-        dj_enabled = self._dj_status_cache.setdefault(
-            ctx.guild.id, await self.config.guild(ctx.guild).dj_enabled()
-        )
+        if ctx.guild.id in self._dj_status_cache:
+            dj_enabled = self._dj_status_cache[ctx.guild.id]
+        else:
+            dj_enabled = self._dj_status_cache.setdefault(
+                ctx.guild.id, await self.config.guild(ctx.guild).dj_enabled()
+            )
         vote_enabled = await self.config.guild(ctx.guild).vote_enabled()
```

Comment:
I'm pretty sure you wanted to switch to call to `all()` here:
```suggestion
        guild_data = await self.config.guild(ctx.guild).all()
        dj_enabled = self._dj_status_cache.setdefault(ctx.guild.id, guild_data["dj_enabled"])
        vote_enabled = guild_data["vote_enabled"]
```

Reply:
> I'm pretty sure you wanted to switch to call to `all()` here:

While I was talking with @Drapersniper about these changes and setdefault, he said this to me:

The idea behind dj_enabled here is to cache is in an attribute
and NEVER call config for it if its in the cache
his "improvement" here calls .all()
which means its always calling the dj_attribute
while i say this 
```
dj_enabled = self._dj_status_cache.setdefault(
            ctx.guild.id, await self.config.guild(ctx.guild).dj_enabled()
        )
```
this right now is broken in audio
because of the way setdefault works
so its not "broken" but its for behaving as intended
the idea here is self._dj_status_cache.setdefault will only call await self.config.guild(ctx.guild).dj_enabled() if the cache for this guild is empty
but thats not how setdefault works unfortunately
which means await self.config.guild(ctx.guild).dj_enabled() ALWAYS gets called